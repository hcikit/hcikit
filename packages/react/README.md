# @hcikit/react

This package is a package that allows you to create and run HCI experiments. Visit hcikit.github.io for more information.
