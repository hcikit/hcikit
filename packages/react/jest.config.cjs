/** @type {import('@ts-jest/dist/types').InitialOptionsTsJest} */
module.exports = {
  preset: "ts-jest",
  testEnvironment: "jsdom",
  moduleNameMapper: {
    "@hcikit/workflow(.*)$": "<rootDir>/../workflow/src$1",
  },
  testMatch: ["./src/**/*.[jt]s?(x)", "**/?(*.)+(spec|test).[t]s?(x)"],
  modulePathIgnorePatterns: ["./dist/", "./node_modules/"],
  testPathIgnorePatterns: ["./dist/", "./node_modules/"],
  transformIgnorePatterns: ["./dist/", "/node_modules/"],
  setupFilesAfterEnv: ["<rootDir>/setupTests.ts"],
  collectCoverageFrom: [
    "src/**/*.{ts,tsx}",
    "!/node_modules/",
    "!src/setupTests.ts",
    "!src/**/*.stories.{js,ts,tsx,jsx}",
  ],
};
