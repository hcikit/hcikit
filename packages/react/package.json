{
  "name": "@hcikit/react",
  "version": "1.0.0-alpha.14",
  "description": "An experiment runner with a set of pre-built experiment components for running hci experiments",
  "author": "blainelewis1",
  "license": "MIT",
  "repository": "hcikit/hcikit",
  "module": "dist/index.js",
  "main": "dist/index.js",
  "type": "module",
  "keywords": [
    "react",
    "hci",
    "experiment"
  ],
  "types": "./dist/index.d.ts",
  "exports": {
    "import": "./dist/index.js",
    "default": "./dist/index.js"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "engines": {
    "node": ">=8",
    "npm": ">=5"
  },
  "scripts": {
    "test": "NODE_OPTIONS=--experimental-vm-modules yarn jest --env=jsdom",
    "test:watch": "NODE_OPTIONS=--experimental-vm-modules yarn jest --watch --env=jsdom --colors --verbose",
    "start": "tsc --watch",
    "build": "tsc --build --verbose",
    "prepack": "yarn run build",
    "storybook": "start-storybook -p 6006",
    "build-storybook": "build-storybook"
  },
  "peerDependencies": {
    "@mui/icons-material": "^5.5.0",
    "@mui/material": "^5.5.0",
    "@mui/styles": "^5.5.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "dependencies": {
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@hcikit/workflow": "workspace:*",
    "core-js": "^3.32.2",
    "lodash-es": "^4.17.21",
    "prop-types": "^15.8.1",
    "react-error-boundary": "^4.0.11",
    "react-is": "^18.2.0",
    "react-markdown": "^8.0.7",
    "serialize-error": "^11.0.2"
  },
  "gitHead": "8d0f2af41121f70996e8b5803d006832b147fd86",
  "devDependencies": {
    "@emotion/react": "^11.11.1",
    "@emotion/styled": "^11.11.0",
    "@mui/icons-material": "^5.14.11",
    "@mui/material": "^5.14.11",
    "@mui/styles": "^5.14.11",
    "@testing-library/jest-dom": "^6.1.3",
    "@testing-library/react": "^14.0.0",
    "@types/jest": "^29.5.5",
    "@types/react": "^18.2.23",
    "jest-diff": "^29.7.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-scripts": "^5.0.1",
    "typescript": "^5.2.2"
  }
}
