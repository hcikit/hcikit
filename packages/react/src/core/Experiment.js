import { throttle } from "lodash-es";
import React, { useEffect, useState } from "react";
import { Provider } from "react-redux";
import PropTypes from "prop-types";

import { configureStore } from "@hcikit/workflow";
import TaskRenderer from "./TaskRenderer";

const STATE_KEY = "HCIKIT_LOGS";

export const loadStateFromSessionStorage = () => {
  try {
    const state = window.sessionStorage.getItem(STATE_KEY);
    if (state) {
      return JSON.parse(state);
    }
  } catch (err) {
    console.error("Failed to load from sessionStorage", err);
  }

  return undefined;
};

export const saveStateToSessionStorage = throttle((state) => {
  try {
    window.sessionStorage.setItem(STATE_KEY, JSON.stringify(state));
  } catch (err) {
    console.error("Failed to save to sessionStorage", err);
  }
}, 3000);

let Experiment = ({
  saveState = saveStateToSessionStorage,
  loadState = loadStateFromSessionStorage,
  ...props
}) => {
  // TODO: maybe we should hide the task registry and instead just pass a list of objects?
  // TODO: not sure how to create different sessions for the same task.

  let [store, setStore] = useState();
  console.log(setStore);
  console.log({ store });
  /* eslint-disable react-hooks/exhaustive-deps */
  useEffect(() => {
    let storedState;

    if (process.env.NODE_ENV !== "development" && loadState) {
      storedState = loadState();
    }

    setStore(
      configureStore({ ...props.configuration, ...storedState }, saveState)
    );
  }, []);

  if (!store) {
    return null;
  }

  return (
    <Provider store={store}>
      <TaskRenderer {...props} />
    </Provider>
  );
};

Experiment.propTypes = {
  configuration: PropTypes.object.isRequired,
  loadState: PropTypes.func,
  saveState: PropTypes.func,
  tasks: PropTypes.objectOf(PropTypes.elementType),
};

export default Experiment;
