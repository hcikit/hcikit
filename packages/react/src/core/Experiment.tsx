import { isEmpty } from "lodash-es";
import React, {
  useState,
  createContext,
  useContext,
  useCallback,
  ElementType,
  useMemo,
  useEffect,
} from "react";

import { serializeError } from "serialize-error";
import TaskRenderer from "./TaskRenderer.js";

import {
  advanceConfiguration,
  logToConfiguration,
  modifyConfiguration,
  Configuration,
  ExperimentIndex,
  UnfilledLog,
  experimentComplete,
  getCurrentIndex,
  TypedConfiguration,
} from "@hcikit/workflow";

import GridLayout from "../GridLayout.js";

import { ErrorBoundary, FallbackProps } from "react-error-boundary";
import { CenteredNicePaper } from "../components/index.js";
import { BasePersistence, StoragePersistence } from "../persistence/index.js";

import { Button, Typography, CircularProgress } from "@mui/material";
import { Task } from "../index.js";

type PropsOfDict<T extends Record<string, React.ComponentType<any>>> = {
  [Task in keyof T]: React.ComponentProps<T[Task]>;
};

export type ConfigurationReact<
  T extends Record<string, React.ComponentType<any>>,
> = TypedConfiguration<PropsOfDict<T>>;

export interface ControlFunctions {
  advance: (index?: ExperimentIndex) => void;
  log: (log: UnfilledLog) => void;
  modify: (
    modifiedConfig: Record<string, unknown>,
    index?: ExperimentIndex
  ) => void;
  persistence?: BasePersistence | null;
}

export const ConfigContext = createContext<Configuration | undefined>(
  undefined
);

export const ConfigMutatorContext = createContext<ControlFunctions | undefined>(
  undefined
);

function useConfiguration(): Configuration {
  const context = useContext(ConfigContext);
  if (context === undefined) {
    throw new Error("useConfig must be used within a ConfigProvider");
  }
  return context;
}

function useExperiment(): ControlFunctions {
  const context = useContext(ConfigMutatorContext);

  if (context === undefined) {
    throw new Error("useConfig must be used within a ConfigMutatorContext");
  }

  return context;
}

export { useConfiguration, useExperiment };

function isPromise<T extends object>(obj: T | Promise<T>): obj is Promise<T> {
  return "then" in obj && typeof obj.then === "function";
}

const Experiment: React.FunctionComponent<{
  persistence?: BasePersistence | null;
  configuration: Configuration | Promise<Configuration>;
  tasks: Record<string, Task<any>>;
  Layout?: ElementType;
  ErrorHandler?: React.ComponentType<FallbackProps>;
  forceRemountEveryTask?: boolean;
  devOptions?: { persistInDevelopment: boolean; useIndexFromUrl: boolean };
}> = ({
  persistence = new StoragePersistence(window.sessionStorage, "HCIKIT_LOGS"),
  configuration: initialConfiguration,
  devOptions = { persistInDevelopment: false, useIndexFromUrl: true },
  ...props
}) => {
  const shouldPersistConfiguration =
    (process.env.NODE_ENV !== "development" ||
      devOptions.persistInDevelopment) &&
    persistence !== null;

  const [configurations, setConfigurations] = useState<{
    loadedConfiguration: Configuration | undefined;
    initialConfiguration: Configuration | undefined;
    doneLoadingPersisted: boolean;
    doneLoadingInitial: boolean;
  }>({
    loadedConfiguration: undefined,
    initialConfiguration: isPromise(initialConfiguration)
      ? undefined
      : initialConfiguration,
    doneLoadingPersisted: !shouldPersistConfiguration,
    doneLoadingInitial: !isPromise(initialConfiguration),
  });

  useEffect(() => {
    async function loadConfigs() {
      if (shouldPersistConfiguration) {
        persistence?.load().then((loadedConfig) => {
          setConfigurations((c) => ({
            ...c,
            loadedConfiguration: loadedConfig,
            doneLoadingPersisted: true,
          }));
        });
      }
      if (isPromise(initialConfiguration)) {
        initialConfiguration.then((loadedConfig) => {
          setConfigurations((c) => ({
            ...c,
            initialConfiguration: loadedConfig,
            doneLoadingInitial: true,
          }));
        });
      }
    }

    loadConfigs();
  }, []);

  if (
    !configurations.doneLoadingInitial ||
    !configurations.doneLoadingPersisted ||
    !configurations.initialConfiguration
  ) {
    return (
      <CenteredNicePaper>
        <Typography>Loading experiment configuration...</Typography>
        <br />
        <CircularProgress size={100} />
      </CenteredNicePaper>
    );
  } else {
    return (
      <ExperimentInner
        {...props}
        devOptions={devOptions}
        initialConfiguration={configurations.initialConfiguration}
        loadedConfiguration={configurations.loadedConfiguration}
        persistence={persistence}
      />
    );
  }
};

const ExperimentInner: React.FC<{
  persistence?: BasePersistence | null;
  initialConfiguration: Configuration;
  loadedConfiguration: Configuration | undefined;
  tasks: Record<string, Task>;
  Layout?: ElementType;
  ErrorHandler?: React.ComponentType<FallbackProps>;
  forceRemountEveryTask?: boolean;
  devOptions: { persistInDevelopment: boolean; useIndexFromUrl: boolean };
}> = ({
  persistence,
  tasks,
  Layout = GridLayout,
  ErrorHandler = DefaultErrorHandler,
  forceRemountEveryTask,
  initialConfiguration,
  loadedConfiguration,
  devOptions,
}) => {
  // TODO: not sure how to create different sessions for the same task. The issue is that they'll be overwritten by the other thing. Maybe we can add a config version or session key or something to it?
  // TODO: using config and configuration is so confusing...
  const [config, setConfig] = useState<Configuration>(() => {
    let configurationToUse: Configuration = initialConfiguration;

    if (
      loadedConfiguration &&
      (process.env.NODE_ENV !== "development" ||
        devOptions.persistInDevelopment)
    ) {
      configurationToUse = loadedConfiguration;
    }

    // This seems hacky. I think I am assigning the initial index to whatever it should be if the experiment just started, or assigning it to the proper one if we haven't started yet.
    configurationToUse = advanceConfiguration(
      configurationToUse,
      getCurrentIndex(configurationToUse)
    );

    return configurationToUse;
  });

  // it might be better to do this as we save state rather than waiting a render, I chose this way because it is easier to implement than adding it to each of the modifying functions below.
  // useEffect(() => {
  //   if (saveState) {
  //     saveState(config);
  //     console.log("saving");
  //   }
  // }, [config, saveState]);

  const advance = useCallback(
    (index?: ExperimentIndex) =>
      setConfig((c) => {
        c = logToConfiguration(c, { type: "END" });
        c = advanceConfiguration(c, index);

        if (!experimentComplete(c)) {
          c = logToConfiguration(c, { type: "START" });
        }

        persistence?.saveThrottled(c);

        return c;
      }),
    [persistence]
  );

  const log = useCallback(
    (log: UnfilledLog): void =>
      setConfig((c) => {
        const newConfig: Configuration = logToConfiguration(c, log);

        persistence?.saveThrottled(c);

        return newConfig;
      }),
    [persistence]
  );

  const modify = useCallback(
    (modifiedConfig: Record<string, unknown>, index?: ExperimentIndex): void =>
      setConfig((c) => {
        const newConfig = modifyConfiguration(c, modifiedConfig, index);

        persistence?.saveThrottled(c);

        return newConfig;
      }),
    [persistence]
  );

  // Kent says this is actually slower than just recreating it... But honestly idk if I trust that. Truth is I should test it, but this does solve the problem of unnecessary rerenders.

  const experiment = useMemo(
    () => ({
      advance: advance,
      log,
      modify,
      persistence,
    }),
    [advance, log, modify]
  );

  useEffect(() => {
    experiment.log({ type: "START" });
    // This is on purpose, we actually never want to rerun this. But also experiment should never actually change.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    window.onbeforeunload = (e) => {
      persistence?.save(config);
      persistence?.flush();
      return "You have some unsaved changes";
    };
  }, []);

  if (isEmpty(config)) {
    return null;
  }

  return (
    <ConfigMutatorContext.Provider value={experiment}>
      <ConfigContext.Provider value={config}>
        <ErrorBoundary
          onError={(error, info) =>
            log({
              type: "ERROR",
              ...serializeError(info),
              ...serializeError(error),
            })
          }
          FallbackComponent={ErrorHandler}
        >
          <Layout>
            {!experimentComplete(config) ? (
              <TaskRenderer
                tasks={tasks}
                forceRemountEveryTask={forceRemountEveryTask}
              />
            ) : (
              <div style={{ gridArea: "task" }}>
                <CenteredNicePaper>
                  <h2>You&apos;ve completed the experiment!</h2>
                  <a
                    download={`${config.participant || "log"}.json`}
                    href={`data:text/json;charset=utf-8,${encodeURIComponent(
                      JSON.stringify(config)
                    )}`}
                  >
                    Download experiment log
                  </a>
                </CenteredNicePaper>
              </div>
            )}
          </Layout>
        </ErrorBoundary>
      </ConfigContext.Provider>
    </ConfigMutatorContext.Provider>
  );
};

const DefaultErrorHandler: React.FunctionComponent<FallbackProps> = ({
  error,
}) => {
  const configuration = useConfiguration();
  return (
    <CenteredNicePaper>
      <div style={{ gridArea: "task" }}>
        <h2>An error occurred in the experiment.</h2>
        <p style={{ fontStyle: "italic" }}>
          ({error.name}: {error.message})
        </p>
        <br />

        <a
          download={`${configuration.participant || "log"}.json`}
          href={`data:text/json;charset=utf-8,${encodeURIComponent(
            JSON.stringify(configuration)
          )}`}
        >
          Download experiment log
        </a>
      </div>
    </CenteredNicePaper>
  );
};

export default Experiment;
