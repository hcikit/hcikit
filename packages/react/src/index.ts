import { PropsWithChildren, ReactElement, Component } from "react";
import { ControlFunctions } from "./core/Experiment.js";
import * as allTasks from "./tasks/index.js";

export * from "./tasks/index.js";
export * from "./components/index.js";

export { allTasks };

export { withGridItem } from "./GridLayout.js";

export { default as GridLayout } from "./GridLayout.js";
export { default } from "./core/Experiment.js";
export * from "./core/Experiment.js";
export {
  BasePersistence,
  StoragePersistence,
  FileSystemPersistence,
} from "./persistence/index.js";

type TaskProps = ControlFunctions;
type StaticProps = {
  layoutProps?: Record<string, any>;
};

export type Task<P = {}> = React.ComponentType<TaskProps & P> & StaticProps;

export interface FunctionTask<P = {}>
  extends React.FunctionComponent<P & ControlFunctions> {
  (
    props: PropsWithChildren<P & ControlFunctions>,
    context?: any
  ): ReactElement<any, any> | null;
}

export abstract class ClassTask<P = {}, S = {}, SS = any> extends Component<
  P & TaskProps,
  S,
  SS
> {}
