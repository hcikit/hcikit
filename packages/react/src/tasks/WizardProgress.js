import React from "react";
import { Stepper, Step, StepLabel } from "@material-ui/core";

import { withGridItem } from "../GridLayout";
import PropTypes from "prop-types";
import { getCurrentIndex } from "@hcikit/workflow";
import { useConfig } from "../core/Experiment";

// TODO: should we use a custom prop like the label or should we consider just using the task and then people can include spaces if they'd like?
// What does include spaces even mean....

// TODO: we could also use a depth parameter here
let WizardProgress = ({ setWorkflowIndex }) => {
  let configuration = useConfig();
  let currentStep = getCurrentIndex(configuration)[0];

  return (
    <Stepper activeStep={currentStep}>
      {configuration.children.map(({ task }, index) => {
        return (
          <Step
            style={{
              cursor:
                process.env.NODE_ENV === "development" ? "pointer" : "default",
            }}
            onClick={() => {
              if (process.env.NODE_ENV === "development") {
                setWorkflowIndex([index]);
              }
            }}
            key={task}
          >
            <StepLabel>{task}</StepLabel>
          </Step>
        );
      })}
    </Stepper>
  );
};

WizardProgress.propTypes = {
  setWorkflowIndex: PropTypes.func,
  configuration: PropTypes.object,
};

WizardProgress = withGridItem(WizardProgress, "header");
export { WizardProgress };

export default WizardProgress;
