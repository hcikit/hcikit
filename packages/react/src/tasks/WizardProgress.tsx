import React from "react";
import { Stepper, Step, StepLabel } from "@material-ui/core";
import { withGridItem } from "../GridLayout";
import {
  getCurrentIndex,
  getConfigurationAtIndex,
  getLeafIndex,
} from "@hcikit/workflow";
import { useConfig, useExperiment } from "../core/Experiment";
import { startCase } from "lodash";

const NoGridWizardProgress: React.FunctionComponent<{ depth: number }> = ({
  depth = 0,
}) => {
  const experiment = useExperiment();
  const configuration = useConfig();
  const pathToIndex = getCurrentIndex(configuration).slice(0, depth);

  let configurationAtIndex = getConfigurationAtIndex(
    pathToIndex,
    configuration
  );

  const currentStep = getLeafIndex(
    getCurrentIndex(configuration),
    configuration
  )[depth];

  if (!configurationAtIndex || !configurationAtIndex.children) {
    console.error(
      "Attempted to render WizardProgress at a depth that does not have any children"
    );
    return null;
  }

  return (
    <Stepper activeStep={currentStep}>
      {configurationAtIndex.children.map(({ task, label }, index) => {
        return (
          <Step
            style={{
              cursor:
                process.env.NODE_ENV === "development" ? "pointer" : "default",
            }}
            onClick={() => {
              if (process.env.NODE_ENV === "development") {
                experiment.setWorkflowIndex([...pathToIndex, index]);
              }
            }}
            key={index}
          >
            <StepLabel>{(label as string) || startCase(task)}</StepLabel>
          </Step>
        );
      })}
    </Stepper>
  );
};

const WizardProgress = withGridItem(NoGridWizardProgress, "header");
export { WizardProgress };

export default WizardProgress;
