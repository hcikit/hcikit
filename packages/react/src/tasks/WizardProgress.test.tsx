import React from "react";
import WizardProgress from "./WizardProgress";
import { render, screen } from "@testing-library/react";
import Experiment, { useExperiment } from "../core/Experiment";
import { Configuration } from "@hcikit/workflow";

let BlankTask: React.FunctionComponent = () => null;

let basicConfig: Configuration = {
  __INDEX__: [0],
  tasks: ["WizardProgress"],
  children: [
    { label: "hello world", task: "BlankTask" },
    { task: "BlankTask" },
    { label: "labels", task: "BlankTask" },
  ],
};

let advancedConfig: Configuration = {
  tasks: ["WizardProgress"],
  depth: 1,

  children: [
    {
      label: "hello world",
      children: [
        { label: "0,0", task: "BlankTask" },
        { label: "0,1", task: "BlankTask" },
      ],
    },
    {
      label: "hello world",
      children: [
        { label: "1,0", task: "BlankTask" },
        { label: "1,1", task: "BlankTask" },
      ],
    },
    { label: "hello world", task: "BlankTask" },
  ],
};

let ClickToAdvance: React.FunctionComponent = () => {
  const experiment = useExperiment();
  return <button onClick={experiment.taskComplete}>advance</button>;
};

describe("WizardProgress", () => {
  it("renders without crashing", () => {
    render(
      <Experiment
        tasks={{
          ClickToAdvance,
          WizardProgress,
          BlankTask,
        }}
        configuration={{ ...basicConfig }}
        saveState={null}
        loadState={null}
      />
    );

    expect(screen.getByText("hello world")).toHaveClass("MuiStepLabel-active");
    expect(screen.getByText("Blank Task")).not.toHaveClass(
      "MuiStepLabel-active"
    );
    expect(screen.getByText("labels")).not.toHaveClass("MuiStepLabel-active");
  });

  it("renders correctly selected task", () => {
    render(
      <Experiment
        tasks={{
          ClickToAdvance,
          WizardProgress,
          BlankTask,
        }}
        configuration={{ ...basicConfig, __INDEX__: [1] }}
        saveState={null}
        loadState={null}
      />
    );

    expect(screen.getByText("hello world")).not.toHaveClass(
      "MuiStepLabel-active"
    );
    expect(screen.getByText("Blank Task")).toHaveClass("MuiStepLabel-active");
    expect(screen.getByText("labels")).not.toHaveClass("MuiStepLabel-active");
  });

  it("renders correct depth parameter", () => {
    render(
      <Experiment
        tasks={{
          ClickToAdvance,
          WizardProgress,
          BlankTask,
        }}
        configuration={{ ...advancedConfig }}
        saveState={null}
        loadState={null}
      />
    );
    expect(screen.getByText("0,0")).toHaveClass("MuiStepLabel-active");
    expect(screen.getByText("0,1")).not.toHaveClass("MuiStepLabel-active");
  });
  it("renders correct depth parameter with second one", () => {
    render(
      <Experiment
        tasks={{
          ClickToAdvance,
          WizardProgress,
          BlankTask,
        }}
        configuration={{ ...advancedConfig, __INDEX__: [1, 0] }}
        saveState={null}
        loadState={null}
      />
    );

    expect(screen.getByText("1,0")).toHaveClass("MuiStepLabel-active");
    screen.getByText("1,1");
  });

  it("renders correctly", () => {
    expect(
      render(
        <Experiment
          tasks={{
            ClickToAdvance,
            WizardProgress,
            BlankTask,
          }}
          configuration={{ ...advancedConfig }}
          saveState={null}
          loadState={null}
        />
      ).asFragment()
    ).toMatchSnapshot();
  });
});
