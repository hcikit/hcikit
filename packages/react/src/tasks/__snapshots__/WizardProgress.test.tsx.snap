// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WizardProgress renders correctly 1`] = `
<DocumentFragment>
  <div
    id="grid-layout"
    style="display: grid; width: 100%; height: 100%; grid-template-columns: 1fr; grid-template-rows: min-content 1fr min-content; grid-template-areas: \\"header\\"
      \\"task\\"
      \\"footer\\";"
  >
    <div
      style="grid-area: header;"
    >
      <div
        class="MuiPaper-root MuiStepper-root MuiStepper-horizontal MuiPaper-elevation0"
      >
        <div
          class="MuiStep-root MuiStep-horizontal"
          style="cursor: default;"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root MuiStepIcon-active"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  1
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiStepLabel-active MuiTypography-body2 MuiTypography-displayBlock"
              >
                0,0
              </span>
            </span>
          </span>
        </div>
        <div
          class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled"
        >
          <span
            class="MuiStepConnector-line MuiStepConnector-lineHorizontal"
          />
        </div>
        <div
          class="MuiStep-root MuiStep-horizontal"
          style="cursor: default;"
        >
          <span
            class="MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled"
          >
            <span
              class="MuiStepLabel-iconContainer"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiStepIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="12"
                />
                <text
                  class="MuiStepIcon-text"
                  text-anchor="middle"
                  x="12"
                  y="16"
                >
                  2
                </text>
              </svg>
            </span>
            <span
              class="MuiStepLabel-labelContainer"
            >
              <span
                class="MuiTypography-root MuiStepLabel-label MuiTypography-body2 MuiTypography-displayBlock"
              >
                0,1
              </span>
            </span>
          </span>
        </div>
      </div>
    </div>
    <button>
      advance
    </button>
  </div>
</DocumentFragment>
`;
